roles/ocm-install-core/tasks/main.yml

```diff
@@ -22,7 +22,46 @@
       spec:
         targetNamespaces:
           - open-cluster-management
-          
+
+- name: Operator Subscription Exists
+  kubernetes.core.k8s_info:
+    api_version: operators.coreos.com/v1alpha1
+    kind: Subscription
+    name: advanced-cluster-management
+    namespace: open-cluster-management
+  register: ocm_operator_sub
+
+- name: Assumed Installing
+  set_fact:
+    upgrading: false
+
+- name: Subscription Exists, but Test if New Version
+  block:
+    - name: Get Currently Installed Subscription
+      set_fact:
+        ocm_installedCSV: "{{ lookup('k8s', kubeconfig=ocm_hub_kubeconfig, api_version='operators.coreos.com/v1alpha1', kind='Subscription', namespace='open-cluster-management', resource_name='advanced-cluster-management') | json_query(query) }}"
+      vars:
+        query: "status.installedCSV"
+
+    - name: Is New Version
+      set_fact:
+        upgrading: "{{ ocm_installedCSV != 'advanced-cluster-management.v' + ocm_version }}"
+  when: ocm_operator_sub.resources | length != 0
+
+- name: "[Install] Set ocm_startingCSV"
+  set_fact:
+    ocm_startingCSV: "advanced-cluster-management.v{{ ocm_version }}"
+  when: not upgrading
+
+- name: "[Upgrade] Set ocm_startingCSV"
+  set_fact:
+    ocm_startingCSV: "{{ ocm_installedCSV }}"
+  when: upgrading
+
+- name: Plan Approval Automatic for Install/Upgrade
+  set_fact:
+    ocm_plan_approval: 'Automatic'
```

Comment:
do we want to update it before create/update subscription?

Reply:
I found during testing, that when I create the subscription and if the plan approval is manual, it will not install or upgrade the operator. Hence, I set it to automatic first, allow the operator to be setup, and then subsequently set it to the user-value in line 134.

Reply:
yes, if manual will need an additional step to approve the installplan

Reply:
and do we want `when: not upgrading` for this one? asking as not sure if upgrading path will use this directly?

Reply:
No, I do not think we should add this condition. In both cases (install and upgrade), the plan approval needs to be Automatic for the install or upgrade to happen. The Upgrade path does use this variable also in the next step (line 65) where the subscription is updated with the new channel and version information.